### ISSUE TITLE
programming error using callLater leads to segmentation fault

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7173]
|<img alt="stefanp's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| stefanp reported|
|-|-|
|Trac ID|trac#372|
|Type|defect|
|Created|2003-10-28 18:52:59Z|

<details><summary>Searchable metadata</summary>

```
trac-id__372 372
type__defect defect
reporter__stefanp stefanp
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1067367179000000 1067367179000000
changetime__1469112404079350 1469112404079350
version__ 
owner__ 
cc__stefanp
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="stefanp's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|stefanp commented|
|-|-|

```
#!html
<pre>
Hey all,

This bug is realy simple:

# 'background' function that does stuff and re-schedules itself
# with a newbie error, adding parenthesis after the function name
# in callLater
def somefunction():
  # do stuff
  reactor.callLater(10, somefunction()) # newbie error

# init the 'background task'
reactor.callLater(10, somefunction()) # same error

The result as you can guess is that somefunction is called recursively,
leading to segmentation fault on my box (linux,python2.3,twisted 1.0.7)

While this is definitely a stupid programming error, I think this is a bug, as a
plain python version of the same concept:

def a():
  a()
a()

brings up a "RuntimeError: maximum recursion depth exceeded"...
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="stefanp's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|stefanp commented|
|-|-|

```
#!html
<pre>
Haha ! Found it, it's not a twisted bug AT ALL
I wasn't able to reproduce it with simpler code ... and then I realised that my
code was using psyco ! So, psyco bug here..

Sorry for the wasted bytes :/ I set status to resolved.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset edcf16c3c61f619c312b1934972d1675bb08d426
```
#!CommitTicketReference repository="" revision="edcf16c3c61f619c312b1934972d1675bb08d426"
Merge pull request #372 from twisted/8604-rodrigc-zope-pydoctor

Author: rodrigc
Reviewer: hawkowl
Fixes: #8604
```
